llvm/lib/Target/PowerPC/PPCMIPeephole.cpp

```diff
@@ -1291,6 +1291,10 @@ bool PPCMIPeephole::simplifyCode() {
           addRegToUpdate(OrigOp1Reg);
           if (MI.getOperand(1).isReg())
             addRegToUpdate(MI.getOperand(1).getReg());
+          if (ToErase && ToErase->getOperand(1).isReg())
+            for (auto UseReg: ToErase->explicit_uses())
+              if (UseReg.isReg())
+                addRegToUpdate(UseReg.getReg());
```

Comment:
We have `recomputeLVForDyingInstr()` to recompute the liveness for  operands of`ToErase` instructions. But this function only does the computation for the defs and the regs added by `addRegToUpdate()`.

For now, it is OK we fix this case by adding the use operands of `ToErase` to the update container at this place. If we continue to meet new issues, we may need to revisit the way we recompute the liveness for `ToErase` instructions.